backend/src/models/user.rs

```diff
@@ -156,75 +167,70 @@ impl users::Model {
         Ok(())
     }
 
-    pub async fn update(
+    pub async fn delete(
         database: &DatabaseConnection,
         user: users::Model,
-        event_config: &EventConfig,
         id: SeaOrmUuid,
-        update_user_json: UpdateUserModel,
     ) -> Result<(), Error> {
-        let updated_user = users::Entity::find_by_id(id)
+        let user_to_delete = users::Entity::find_by_id(id)
             .one(database)
             .await?
             .ok_or(Error::UserNotFound)?;
 
-        if user.roles <= updated_user.roles {
+        if user.roles <= user_to_delete.roles {
             return Err(Error::UserMustHaveHigherRoleThanAffectedUser);
         }
 
-        let mut active_user: ActiveModel = updated_user.into();
+        user_to_delete.delete(database).await?;
 
-        if let Some(username) = update_user_json.username {
-            active_user.username = Set(username);
-        }
+        Ok(())
+    }
 
-        if let Some(email) = update_user_json.email {
-            active_user.email = Set(email);
-        }
+    pub async fn update(
+        database: &DatabaseConnection,
+        user: users::Model,
+        update_user_model: UpdateUserModel,
```

Comment:
`UpdateUserModel` shouldn't be used outside update route. Here accept `UpdatableModel`